dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -82,35 +155,93 @@ HRESULT CALLBACK DrivesContextMenuCallback(IShellFolder *psf,
         _ILFreeaPidl(apidl, cidl);
         return E_FAIL;
     }
+    nDriveType = GetDriveTypeA(szDrive);
+    GetVolumeInformationA(szDrive, NULL, 0, NULL, NULL, &dwFlags, NULL, 0);
+
+    static UINT s_idCmdFormat = 0x7FFF;
+    static UINT s_idCmdEject = 0x7FFF;
+    static UINT s_idCmdDisconnect = 0x7FFF;
 
     if (uMsg == DFM_MERGECONTEXTMENU)
     {
         QCMINFO *pqcminfo = (QCMINFO *)lParam;
-        DWORD dwFlags;
 
-        if (GetVolumeInformationA(szDrive, NULL, 0, NULL, NULL, &dwFlags, NULL, 0))
+        s_idCmdFormat = 0x7FFF;
+        s_idCmdEject = 0x7FFF;
+        s_idCmdDisconnect = 0x7FFF;
```

Comment:
It would be preferable if we avoided using static variables in this class. It can be improved if IDS_FORMATDRIVE always gets the pqcminfo->idCmdFirst+1 id, IDS_EJECT gets the pqcminfo->idCmdFirst+ 2 and IDS_DISCONNECT pqcminfo->idCmdFirst+ 3 and right before DFM_MERGECONTEXTMENU returns, always set pqcminfo->idCmdFirst+=3.
This way when handling DFM_INVOKECOMMAND you can use a switch(wParam) to check if wParam is either DFM_INVOKECOMMAND, 1,2 or 3.